src/libraries/Common/src/Roslyn/SyntaxValueProvider_ForAttributeWithSimpleName.cs

```diff
@@ -6,17 +6,39 @@
 using System.Collections.Immutable;
 using System.Diagnostics;
 using System.Linq;
+using System.Runtime.CompilerServices;
 using System.Threading;
 
 using Microsoft.CodeAnalysis;
-
+using Microsoft.CodeAnalysis.CSharp;
 using Aliases = System.Collections.Generic.ValueListBuilder<(string aliasName, string symbolName)>;
 
 namespace Microsoft.CodeAnalysis.DotnetRuntime.Extensions;
 
 internal static partial class SyntaxValueProviderExtensions
 {
-    // private static readonly ObjectPool<Stack<string>> s_stackPool = new(static () => new());
+    /// <summary>
+    /// Information computed about a particular tree.  Cached so we don't repeatedly recompute this important
+    /// information each time the incremental pipeline is rerun.
+    /// </summary>
+    private sealed record SyntaxTreeInfo(SyntaxTree Tree, bool ContainsGlobalAliases, bool ContainsAttributeList);
```

Comment:
recorded information per tree tree.  used so we don't have to recompute this information (Which inovlved parsing) on trees that didn't change.